chromium_src/net/url_request/redirect_util.cc

```diff
@@ -26,12 +26,15 @@ void RedirectUtil::UpdateHttpRequest(
                                  modified_headers,
                                  request_headers,
                                  should_clear_upload);
-  // Hack for dropping referrer at the network layer.
+  // Hack for capping referrers at the network layer.
   if (removed_headers) {
     if (removed_headers->end() != std::find(removed_headers->begin(),
                                             removed_headers->end(),
-                                            "X-Brave-Clear-Referer")) {
-      const_cast<RedirectInfo&>(redirect_info).new_referrer.clear();
+                                            "X-Brave-Cap-Referer")) {
+      url::Origin referrer = url::Origin::Create(
```

Comment:
Does this take into account HTTPS -> HTTP transitions where the referrer needs to be empty?

Reply:
huh, good catch - should be fixed now